src/app/encode/sorting-task-component/sorting-task.component.ts

```diff
@@ -27,25 +29,38 @@ export class EncodeSortingTaskComponent implements OnInit, OnExit {
 
   constructor(private _userService: EncodeUserService,
     private _router: Router,
-    private _route: ActivatedRoute,)
+    private _route: ActivatedRoute,
+    private _dialog: MatDialog)
   {
   }
 
+  onExit(): boolean | UrlTree | Observable<boolean | UrlTree> | Promise<boolean | UrlTree> {
+    const exitDialogRef = this._dialog.open(ExitConfirmComponent);
+    exitDialogRef.afterClosed().subscribe(this._exitDialogClosed$);
+    return exitDialogRef.afterClosed().toPromise<boolean>();
+  }
+
+  private _exitDialogClosed$ = async (response: boolean): Promise<boolean> => {
+    if (response == true){ 
+      this._userService.abandonTest();
+      return true;
```

Comment:
Probar, creo que devolviendo el UrlTree te navega
```suggestion
      return true;
```

```suggestion
      const redirectUrl: UrlTree = this._router.parseUrl('/');
      return redirectUrl;
```

[UrlTree Angular Docs](https://angular.io/api/router/UrlTree)